src/NexusMods.DataModel/Loadouts/DiskStateRegistry.cs

```diff
@@ -20,16 +20,21 @@ public class DiskStateRegistry : IDiskStateRegistry
 {
     private readonly ILogger<DiskStateRegistry> _logger;
     private readonly IDataStore _dataStore;
+    private readonly IDictionary<GameInstallation, IId> _lastAppliedRevisionDictionary = new Dictionary<GameInstallation, IId>();
+    private readonly Subject<(GameInstallation gameInstallation, IId loadoutRevision)> _lastAppliedRevisionSubject = new();
```

Comment:
This can be simplified using a `SourceCache`.

Reply:
Not as easily as I would like since the revision Id doesn't have a gameInstallation field and getting it from the revision id requires going through the LoaodutRegistry. 
Considering that I don't think it's worth it for now. Current code is pretty simple for now.
I'll keep it in mind though if code gets more complex later